lib/puppet/data_providers.rb

```diff
@@ -4,33 +4,27 @@ def self.assert_loaded
     unless @loaded
       require 'puppet/pops'
       require 'puppet/data_providers/data_adapter'
+      require 'puppet/data_providers/lookup_adapter'
     end
     @loaded = true
   end
 
+  # @deprecated use `lookup_adapter(lookup_invocation).lookup` instead
   def self.lookup_in_environment(name, lookup_invocation, merge)
-    assert_loaded()
-    adapter = DataAdapter.adapt(Puppet.lookup(:current_environment))
-    adapter.env_provider.lookup(name, lookup_invocation, merge)
+    Puppet.deprecation_warning('The method Puppet::DataProviders.lookup_in_environment is deprecated and will be removed in the next major release of Puppet.')
```

Comment:
The deprecation needs a ticket targeting Puppet 5.0.0 so we do not forget to remove the deprecated methods.